.editorconfig

```diff
@@ -0,0 +1,506 @@
+# Docs: https://docs.microsoft.com/en-us/visualstudio/ide/editorconfig-code-style-settings-reference
```

Comment:
Are all these options needed? Seems like many of they are default, just feel like it is a bit overwhelming. But if this is what is needed to have good formatting and linting, sure.

Reply:
I am not sure right now which ones are still default, but in my opinion they are nicely organized and have expressive names such that it will be easy to modify should we find that we have different needs or preferences in the future.
It also gives us an overview of all the options that we have available for controlling the code style and formatting rules